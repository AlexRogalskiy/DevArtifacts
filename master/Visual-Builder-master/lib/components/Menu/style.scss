@import '../../helpers/colors';
@import '../../helpers/fonts';
@import '../../helpers/z-index';

.menu {
  z-index: map-get($z-index-map, menu);
  display: none;
  background: $white;
  width: 200px;
  border-radius: 2px;
  box-shadow: inset 0 0 0 1px rgba($black, 0.1), 0 2px 8px rgba($black, 0.1);

  &--active {
    display: block;
  }

  &__group {
    border-bottom: 1px solid $border;
    margin: 0;
    padding: 10px 0;

    &:last-child {
      border: 0;
    }

    &-content {
      margin: 0;
    }
  }

  &__title,
  &__item {
    display: block;
    width: 100%;
    height: 30px;
    padding: 0 20px;
    box-sizing: border-box;
    font-weight: $font-semibold;
    line-height: 30px;
  }

  &__title {
    color: $text-secondary;
    font-size: 0.625rem;
    text-transform: uppercase;
  }

  &__item {
    display: flex;
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
    width: calc(100% - 2px);
    border-radius: 0;
    margin: 0 1px;
    padding: 0 20px;
    box-sizing: border-box;
    color: $text-primary;
    font-size: 0.75rem;
    text-decoration: none;
    cursor: pointer;

    &:hover {
      background: $light-hover;
    }
  }
}
