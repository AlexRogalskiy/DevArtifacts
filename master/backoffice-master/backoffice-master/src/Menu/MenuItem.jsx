import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import {
  ListItem,
  ListItemIcon,
  ListItemText,
  withStyles,
} from '@material-ui/core'

const styles = {
  isDisabled: {
    opacity: 0.5,
    pointerEvents: 'none',
  },
}

const MenuItem = ({
  redirectTo,
  url,
  title,
  icon: Icon,
  isDisabled,
  classes,
}) => (
  <ListItem
    button
    onClick={() => redirectTo(url)}
    className={classNames({
        [classes.isDisabled]: isDisabled,
      })}
  >
    {Icon ? (
      <ListItemIcon>
        <Icon />
      </ListItemIcon>
      ) : null}
    <ListItemText primary={title} />
  </ListItem>
)

MenuItem.propTypes = {
  redirectTo: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  isDisabled: PropTypes.bool,
  icon: PropTypes.func,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

MenuItem.defaultProps = {
  isDisabled: false,
  icon: null,
}

export default withStyles(styles)(MenuItem)
