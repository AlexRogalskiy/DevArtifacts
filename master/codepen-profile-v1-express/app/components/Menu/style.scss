/* Menu */
.menu {
  z-index: 10;
  position: absolute;
  top: 100%;
  background: #FFF;
  width: 180px;
  border-radius: 2px;
  box-shadow: inset 0 0 0 1px #EEE, 0 2px 8px rgba(#000, 0.1);
  padding: 8px 0;
  box-sizing: border-box;

  &,
  &--hidden {
    display: none;
  }

  &--visible {
    display: block;
  }

  &__item {
    font-size: 0.75rem;
  }

  &__link {
    display: block;
    width: 100%;
    box-sizing: border-box;
    padding: 8px 20px;
    cursor: pointer;

    &:hover {
      background: #F2F2F2;
    }
  }
}