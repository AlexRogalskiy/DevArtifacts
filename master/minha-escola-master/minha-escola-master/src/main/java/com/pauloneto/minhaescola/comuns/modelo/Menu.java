package com.pauloneto.minhaescola.comuns.modelo;

import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

import javax.persistence.*;

/**
 * Entity implementation class for Entity: Menu
 *
 */
@Entity
@Table(name="tb_menu")
public class Menu implements Serializable {

	
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(name="id_menu")
	private Integer idMenu;
	
	@OneToMany(mappedBy="menu",
			   cascade=CascadeType.ALL,
			   fetch=FetchType.EAGER)
	private Set<SubMenu> subMenus;
	
	public Menu(){
		super();
		this.subMenus = new HashSet<SubMenu>();
	}
	
	/**
	 * @return the idMenu
	 */
	public Integer getIdMenu() {
		return idMenu;
	}
	
	/**
	 * @param idMenu the idMenu to set
	 */
	public void setIdMenu(Integer idMenu) {
		this.idMenu = idMenu;
	}

	/**
	 * @return the subMenus
	 */
	public Set<SubMenu> getSubMenus() {
		return subMenus;
	}

	/**
	 * @param subMenus the subMenus to set
	 */
	public void setSubMenus(Set<SubMenu> subMenus) {
		this.subMenus = subMenus;
	}
	
	public void addSubMenu(SubMenu subMenu){
		subMenu.setMenu(this);
		this.subMenus.add(subMenu);
	}
}
