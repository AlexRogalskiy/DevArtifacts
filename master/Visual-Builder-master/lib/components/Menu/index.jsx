/**
 * External dependencies
 */
import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';
import noop from 'lodash/noop';

/**
 * Internal dependencies
 */
import './style.scss';
import Button from '../Button';

/**
 * <Menu />
 */
class Menu extends Component {
  _renderMenuGroup(group, i) {
    return (
      <dl className="menu__group" key={`menu-group-key-${i}`}>
        {group.title ? <dt className="menu__title">{group.title}</dt> : false}
        <dd className="menu__group-content">
          {group.items.map(this._renderMenuGroupItem, this)}
        </dd>
      </dl>
    );
  }

  _renderMenuGroupItem(item, i) {
    if (item.to) {
      return (
        <Button
          className="menu__item"
          key={`menu-group-key-${i}`}
          onClick={() => this.props.changeMenu(item.to)}
        >
          <span>{item.name}</span>
          <i className="ion ion-arrow-right-b" />
        </Button>
      );
    }

    return <Button className="menu__item" key={`menu-item-key-${i}`}>{item.name}</Button>;
  }

  render() {
    const classes = classNames({
      menu: true,
      'menu--active': this.props.status,
    }, this.props.className);

    return (
      <nav className={classes}>
        {this.props.childrenLocation === 'top' ? this.props.children : false}
        {
          this.props.items ?
            this.props.items[this.props.view || 0].map(this._renderMenuGroup, this) : false
        }
        {
          this.props.childrenLocation === 'bottom' ? this.props.children : false ||
          !this.props.childrenLocation ? this.props.children : false
        }
      </nav>
    );
  }
}

Menu.defaultProps = {
  changeMenu: noop,
  children: undefined,
  childrenLocation: undefined,
  className: undefined,
  items: undefined,
  status: false,
  view: undefined,
};

Menu.propTypes = {
  changeMenu: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  childrenLocation: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
  ]),
  className: PropTypes.string,
  items: PropTypes.arrayOf(
    PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string,
        items: PropTypes.arrayOf(
          PropTypes.shape({
            action: PropTypes.func.isRequired,
            name: PropTypes.string.isRequired,
            to: PropTypes.number,
          }),
        ).isRequired,
      }),
    ),
  ),
  status: PropTypes.bool,
  view: PropTypes.number,
};

export default Menu;
