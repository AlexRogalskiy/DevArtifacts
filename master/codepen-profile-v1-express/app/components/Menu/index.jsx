/**
 * External dependencies
 */
import React, { Component, PropTypes } from 'react';
import cx from 'classnames';

/**
 * Internal dependencies
 */
import './style.scss';

/**
 * Local variables
 */
const propTypes = {
  className: PropTypes.string,
  items: PropTypes.array,
  style: PropTypes.object,
};

const defaultProps = {};

/**
 * <Menu />
 */
class Menu extends Component {
  _renderItem(item, key) {
    return (
      <li
        key={`menu-item-${key}`}
        className="menu__item"
      >
        <a
          className="menu__link"
          href={item.url}
          onClick={item.onClick}
        >
          {item.value}
        </a>
      </li>
    );
  }

  render() {
    let classes = cx('menu', this.props.className);
    const items = this.props.items || [];

    return (
      <div
        className={classes}
        style={this.props.style}
      >
        <ul>
          {items.map(this._renderItem.bind(this))}
        </ul>
      </div>
    );
  }
}

Menu.propTypes = propTypes;
Menu.defaultProps = defaultProps;

export default Menu;
