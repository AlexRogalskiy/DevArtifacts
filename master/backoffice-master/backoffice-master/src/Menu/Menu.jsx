import React from 'react'
import PropTypes from 'prop-types'
import {
  Divider,
  List,
  withStyles,
} from '@material-ui/core'

import MenuItem from './MenuItem'

const styles = theme => ({
  root: {
    width: '100%',
    maxWidth: 360,
    background: theme.palette.background.paper,
  },
})

const Menu = ({ data, redirectTo, classes }) => (
  <div className={classes.root}>
    <List>
      {data.map((item) => {
        switch (item.type) {
          case 'divider':
            return <Divider key={`menu-${(Math.random() * 100).toFixed(5)}`} />

          default:
            return (
              <MenuItem
                key={`menu-${(Math.random() * 100).toFixed(5)}`}
                redirectTo={redirectTo}
                url={item.url}
                title={item.title}
                isDisabled={item.isDisabled}
                icon={item.icon}
              />
            )
        }
      })}
    </List>
  </div>
)

Menu.propTypes = {
  redirectTo: PropTypes.func.isRequired,
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

export default withStyles(styles)(Menu)
