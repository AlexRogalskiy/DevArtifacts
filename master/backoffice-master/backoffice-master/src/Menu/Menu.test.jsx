import React from 'react'
import { mount } from 'enzyme'

import Icon from '@material-ui/icons/AccessAlarm'

import Menu from './'

it('renders correctly', () => {
  const tree = mount((
    <Menu
      data={[{
        title: 'Title',
        url: '/',
        icon: Icon,
      }]}
      redirectTo={() => { }}
    />
  ))

  expect(tree).toMatchSnapshot()
})
