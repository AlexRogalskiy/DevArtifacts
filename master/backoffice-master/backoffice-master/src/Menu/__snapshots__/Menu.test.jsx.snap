// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Menu-root-1"
>
  <ul
    className="MuiList-root-2 MuiList-padding-3"
  >
    <div
      className="MuiButtonBase-root-18 MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-14 MuiListItem-button-15"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-22 MuiListItemIcon-root-21"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM12.5 8H11v6l4.75 2.85.75-1.23-4-2.37V8zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
        />
      </svg>
      <div
        className="MuiListItemText-root-29"
      >
        <span
          className="MuiTypography-root-35 MuiTypography-subheading-42 MuiListItemText-primary-32"
        >
          Title
        </span>
      </div>
      <span
        className="MuiTouchRipple-root-60"
      />
    </div>
  </ul>
</div>
`;
